### PR TITLE
USB4SAP : initial pull 10/14/22

### PR DESCRIPTION
**Pull request by ecsadm**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/USB4SAP/Readme.md [hunk 1]
```diff
@@ -1 +1,26 @@
...
+## Supported Operations
+This query supports GET operation from SAP backend to provision data from tables, views, reports, queries, ALV, CDS, or any other data container.
+
+
+## Obtaining Credentials
+This API uses Basic Authentication
+
+## Known Issues and Limitations
+REST API supports data limits as specified under REST API protocols
+Large datasets extraction may result in time outs as configured in the REST API systems
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/Readme.md:21]: @ecsadm all files should be free of grammatical errors and typos. May I ask you please to change "time outs" to "timeouts".

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "email": "hello@ecoservity.com" 
+        } 
+    }, 
+    "host": "https://ecs-apim1.azure-api.net/usb4sap_live_cache-v2/", 
+    "paths": { 
+        "/sap/opu/odata/ECOS/OBJ2CLOUD_V2_SRV/ET_DatasetSet": { 
+            "GET": { 
+                "summary": "GET call to extract metadata & data from SAP data objects", 
+                "operationId": "GetCallExtractMetadata", 
+                "description": "The GET call extracts data alongwith its metadata from SAP, based on the selection criteria by user or background scheduled job",
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:19]: @ecsadm all Operations descriptions should end with a period. No period required only if there's a single word.
Please see this useful [documentation](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)
all files should be free of grammatical errors and typos. May I ask you please to change "alongwith" to "along with".

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            "name": "Ecoservity", 
+            "url": "https://www.ecoservity.com/#contact", 
+            "email": "hello@ecoservity.com" 
+        } 
+    }, 
+    "host": "https://ecs-apim1.azure-api.net/usb4sap_live_cache-v2/", 
+    "paths": { 
+        "/sap/opu/odata/ECOS/OBJ2CLOUD_V2_SRV/ET_DatasetSet": { 
+            "GET": { 
+                "summary": "GET call to extract metadata & data from SAP data objects",
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:17]: @ecsadm special characters in not allowed in the operation summary. pleases remove the "&".

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                "summary": "GET call to extract metadata & data from SAP data objects", 
+                "operationId": "GetCallExtractMetadata", 
+                "description": "The GET call extracts data alongwith its metadata from SAP, based on the selection criteria by user or background scheduled job", 
+                "parameters": [ 
+                    { 
+                        "name": "$filter", 
+                        "in": "query", 
+                        "required": true, 
+                        "type": "string", 
+                        "x-example": "ObjectName eq 'T001W' and ObjectType eq 'T' and Delimiter eq '|' and ImportGeneric eq 'ALVV-MG' and DataDest eq '0:0|' "
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:26]: @ecsadm all parameters should have 
"x-ms-summary": "Your Summary"
"description": "Your description"
Please see our docs [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters)

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                        "required": true, 
+                        "type": "string", 
+                        "x-example": "ObjectName eq 'T001W' and ObjectType eq 'T' and Delimiter eq '|' and ImportGeneric eq 'ALVV-MG' and DataDest eq '0:0|' " 
+                    }, 
+                    { 
+                        "name": "$format", 
+                        "in": "query", 
+                        "required": false, 
+                        "type": "string", 
+                        "x-example": "json"
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:33]: @ecsadm all parameters should have 
"x-ms-summary": "Your Summary"
"description": "Your description"
Please see our docs [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters)

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        "version": "1.0", 
+        "description": "Universal Synch Bridge for SAP: One access point for all SAP integration, analytics and UI needs. Simplify modern analytics, apps and automations. Provision SAP data, insights and automation On-demand.", 
+        "contact": { 
+            "name": "Ecoservity", 
+            "url": "https://www.ecoservity.com/#contact", 
+            "email": "hello@ecoservity.com" 
+        } 
+    }, 
+    "host": "https://ecs-apim1.azure-api.net/usb4sap_live_cache-v2/", 
+    "paths": {
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:14]: @ecsadm schemes property is required. Please add
 "schemes": ["https"],

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                    } 
+                } 
+            } 
+        }, 
+        "definitions": {} 
+    }, 
+    "x-ms-connector-metadata": [ 
+        { 
+            "propertyName": "Website", 
+            "propertyValue": "www.ecoservity.com"
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:60]: @ecsadm this should be a valid url. please change to "https://www.ecoservity.com"

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        "definitions": {} 
+    }, 
+    "x-ms-connector-metadata": [ 
+        { 
+            "propertyName": "Website", 
+            "propertyValue": "www.ecoservity.com" 
+        }, 
+        { 
+            "propertyName": "Privacy policy", 
+            "propertyValue": "NA"
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:64]: @ecsadm this should be a valid url.

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+            "propertyName": "Website", 
+            "propertyValue": "www.ecoservity.com" 
+        }, 
+        { 
+            "propertyName": "Privacy policy", 
+            "propertyValue": "NA" 
+        }, 
+        { 
+            "propertyName": "Categories", 
+            "propertyValue": "B2B Analytics"
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:68]: @ecsadm this is not a valid category. you can choose one of the below.
Available categories are: [AI, Business Management, Business Intelligence, Collaboration, Commerce, Communication, Content and Files, Finance, Data, Human Resources, Internet of Things, IT Operations, Lifestyle and Entertainment, Marketing, Productivity, Sales and CRM, Security, Social Media, Website]

File: certified-connectors/USB4SAP/apiProperties.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
+{
+    "properties": {
+        "connectionParameters": {
+            "api_key": {
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiProperties.json:4]: @ecsadm to have a basic authentication type you can just have this connectionParameters as empty property. 
Please change to "connectionParameters": {}

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by ecsadm** [comment]
> 

@microsoft-github-policy-service agree [company="{Ecoservity}"]

File: certified-connectors/USB4SAP/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+                    "500": { 
+                        "description": "Application Error" 
+                    }, 
+                    "default": { 
+                        "description": "Swagger generated by Ecoservity" 
+                    } 
+                } 
+            } 
+        }, 
+        "definitions": {}
```
Comment (Amjed-Ayoub) [inline certified-connectors/USB4SAP/apiDefinition.swagger.json:61]: @ecsadm please move the definitions out of the paths scope.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Review by Amjed-Ayoub — APPROVED** [review]

**Comment by Amjed-Ayoub** [comment]
Hello @ecsadm,
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-updates) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector. 
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Comment by ecsadm** [comment]
Thanks Amjed, congratulations to you as well.
We will work on this & revert soon with actions.





Thanks a ton,
Anupam Jaiswal
M: ***@***.******@***.***> | T: +1.2145199601
https://www.ecoservity.com/






From: Amjed Ayoub ***@***.***>
Date: Tuesday, October 18, 2022 at 17:03
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Anupam Jaiswal ***@***.***>, Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] USB4SAP : initial pull 10/14/22 (PR #2059)

Hello @ecsadm<https://github.com/ecsadm>,
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please submit the files to ISV Studio<https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio> or update your submission<https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-updates> using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please make sure that the icon meets all the requirements<https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector>, that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can find an example of intro.md file on this page<https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio> at step 6.
Please Create an environment in the Preview region<https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region>. You will use this environment later to test your connector<https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector> after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**